### DISCUSSION TITLE
Migrating from --normal_map to --normal_mode with v4.1.0-rc1

### DISCUSSION THREAD

**Discussion by donmccurdy** [KhronosGroup/KTX-Software#601] [Q&A] [answered]
In the latest release candidate, [v4.1.0-rc1](https://github.com/KhronosGroup/KTX-Software/releases/tag/v4.1.0-rc1), I've noticed that the previous `--normal_map` flag has been replaced by `--normal_mode`. My understanding is that the former would optimize encoding of 3-channel normal maps (which I understand is quite lossy for ETC1S), while the latter also converts the input to a two-channel X+Y normal map.

The glTF-Transform CLI uses this utility when processing glTF files, and currently glTF does not allow two-channel X+Y normal maps. How would you recommend optimizing the encoder for normal maps, while keeping the normal map in a three-channel format? I would guess there will be software other than just glTF-related tools that also continue to make some use of three-channel normal maps, as much as it'd be great to see more adoption of X+Y here. Thanks!

**Comment by MarkCallow** [comment, marked as answer]
The only optimization done by the BasisU encoders for normal maps is to turn off RDO in the ETC1S/BasisLZ encoder. There are 2 ways to get this optimization for 3-component normal maps.

```
--input_swizzle rgb1 --normal_mode
```

or

```
--no_endpoint_rdo --no_selector_rdo
```

The former is preferable as it is encoder agnostic and will pick up any normal map optimizations added to the BasisU encoders in future. The latter only has an effect when encoding to ETC1S/BasisLZ.

**Comment by donmccurdy** [comment]
Thank you! The addition of `--input_swizzle rgb1` is working for me in both modes.

**Comment by MarkCallow** [comment]
I'll add a note in the documentation.

I'll convert this to a question/answer discussion.
